\documentclass{article}
\usepackage{fullpage}

\begin{document}
<<set-options, echo=FALSE, cache=FALSE>>=
opts_chunk$set(echo = FALSE, fig.path = 'figure/', dev='tikz', fig.align='center', fig.show='hold')
read_chunk('1_load.R')
@

\title{Microarray Analysis}
\date{\today}
\maketitle

\section{Introduction}
Contigs for this report were selected by looking for perfect matches against 
the \textit{Eurydice pulchra} mitochondrial genome. A perfect match is defined
as having an e value of 0.0.

<<load, echo=FALSE>>=
@

<<ext_code1, echo=FALSE>>=
read_chunk('1.5_functions.R')
@

<<func1, echo=FALSE>>=
@

<<ext_code2, echo=FALSE>>=
read_chunk('2_clean.R')
@

<<clean, echo=FALSE>>=
@

<<ext_code3, echo=FALSE>>=
read_chunk('3_functions.R')
@

<<func2, echo=FALSE>>=
@

<<ext_code4, echo=FALSE>>=
read_chunk('4_do.R')
@

<<filter, echo=FALSE>>=
@
list of included mitochondrial genes (and their contigs)

Graph over whole time course (both with and without rrnL) ordered for circadian and ordered for tidal
comment on not being able to reconcile those two 

Used t test to compare mean expression for either HW vs LW or Day vs Night

table of p values one for tidal, one for circadian

graphs of change


\end{document}